### PR TITLE
(0.105.1) Add density and gravity_wave_speed to PerturbationAdvection

### PR DESCRIPTION
**Pull request by glwagner**
## Summary

Extends `PerturbationAdvection` with two optional keyword arguments for models with density-weighted prognostic variables (e.g., anelastic atmospheric models):

- **`density`**: when provided, converts density-weighted fields (ρψ) to intensive fields (ψ) before radiation and back after. Fixes the dimensional inconsistency when applying `PerturbationAdvection` directly to prognostic variables like ρu, ρθ, ρq.

- **`gravity_wave_speed`**: additional phase speed added to the advective velocity, following Klemp & Wilhelmson (1978). Useful for momentum fields where gravity waves propagate faster than the mean flow.

Both default to `nothing` and `0` respectively, so this is **fully backward compatible**.

Also adds `_fill_*_halo!` methods for `Center`-located fields, enabling `PerturbationAdvection` to be used on scalar fields (tracers, ρθ, etc.) in addition to `Face`-located velocity/momentum fields.

## Motivation

This is motivated by [NumericalEarth/Breeze.jl#494](https://github.com/NumericalEarth/Breeze.jl/pull/494), which would otherwise have to maintain a separate `PerturbationMomentumAdvection` scheme with ~270 lines of duplicated radiation logic. With this upstream change, Breeze can use `PerturbationAdvection` directly and delete the duplicate code.

## Test plan

- [x] Existing `PerturbationAdvection` API is unchanged (new kwargs have defaults)
- [x] Existing boundary condition integration tests pass
- [x] Test with density-weighted fields in Breeze.jl

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/BoundaryConditions/perturbation_advection.jl [hunk 1]
```diff
@@ -1,18 +1,22 @@
 using Oceananigans.Operators: Δxᶠᶜᶜ, Δyᶜᶠᶜ, Δzᶜᶜᶠ
 using Oceananigans: defaults
 
-struct PerturbationAdvection{FT}
-    inflow_timescale :: FT
-   outflow_timescale :: FT
+struct PerturbationAdvection{FT, D}
+     inflow_timescale :: FT
+    outflow_timescale :: FT
```
Comment (glwagner) [inline src/BoundaryConditions/perturbation_advection.jl]:
```suggestion
    inflow_timescale :: FT
    outflow_timescale :: FT
```

**Comment by glwagner** [comment]
i think tests will pass but CI is being flaky now

File: Project.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name = "Oceananigans"
 uuid = "9e8cae18-63c1-5223-a75c-80ca9d6e9a09"
-version = "0.105.0"
+version = "0.105.1"
```
Comment (glwagner) [inline Project.toml:3]:
```suggestion
version = "0.105.0"
```

File: src/BoundaryConditions/perturbation_advection.jl [hunk 3]
```diff
@@ -53,128 +57,232 @@ to point into the domain.
...
+                               density = nothing)
     inflow_timescale = convert(FT, inflow_timescale)
     outflow_timescale = convert(FT, outflow_timescale)
-    return PerturbationAdvection(inflow_timescale, outflow_timescale)
+    gravity_wave_speed = convert(FT, gravity_wave_speed)
+    return PerturbationAdvection(inflow_timescale, outflow_timescale, gravity_wave_speed, density)
 end
 
+# Backward compatibility: support old 2-positional-arg constructor
+PerturbationAdvection(inflow_timescale::Number, outflow_timescale::Number) =
```
Comment (jagoosw) [inline src/BoundaryConditions/perturbation_advection.jl]: I'm not sure this type annotation is necessary

https://github.com/jrevels/YASGuide/blob/bb636744bb117a6e1f115789a1e0f5034b7360ae/README.md?plain=1#L80

Comment (glwagner) [reply src/BoundaryConditions/perturbation_advection.jl]: agree, its just the 2 arguments that are needed

File: src/BoundaryConditions/perturbation_advection.jl [hunk 4]
```diff
@@ -53,128 +82,232 @@ to point into the domain.
...
```
Comment (glwagner) [inline src/BoundaryConditions/perturbation_advection.jl]:
```suggestion
# Support 2-positional-arg constructor
PerturbationAdvection(inflow_timescale, outflow_timescale) =
```

File: src/BoundaryConditions/perturbation_advection.jl [hunk 5]
```diff
@@ -1,45 +1,74 @@
...
+```math
+\\psi'^{B, n+1} = \\frac{\\psi^{B, n} + \\tilde{U} \\, \\psi'^{A, n+1} - \\bar{\\psi}^{n+1}}
+                       {1 + \\tilde{U} + \\Delta t / \\tau},
+```
+
+where ``\\tilde{U} = \\bar{\\psi} \\, \\Delta t / \\Delta x``. Then ``\\psi^{B, n+1}`` is:
+
+```math
+\\psi^{B, n+1} = \\frac{\\psi^{B, n} + \\tilde{U} \\, \\psi^{A, n+1} + \\bar{\\psi}^{n+1} \\tilde{\\tau}}
+                      {1 + \\tilde{\\tau} + \\tilde{U}}
```
Comment (tomchor) [inline src/BoundaryConditions/perturbation_advection.jl]: Do we want to switch to latex for the math here? It doesn't render on the REPL (afaik), which is I think how most people check docstrings.

Comment (glwagner) [reply src/BoundaryConditions/perturbation_advection.jl]: ok, switching to unicode. the main thing i wanted to hcnage was to use the math block. but the unicode should be auotmaitcally converted to latex for the docs
